src/domain/flow-system/src/entities/shared/dataset_flow_type.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq)]
+pub enum DatasetFlowType {
+    Update,
```

Comment:
Would it make sense to break this flow into `Ingest` and `ExecuteQuery`? I feel like `Update` is too generic and it would be very nice for these flow types to correspond to the names of operations the engines will be executing.

Reply:
Even if all implementation will be share for these types of flow, having an ability to query flow system e.g. to see how much compute time we're spending on ingest vs. derivative updates could be nice to have.

Reply:
I've separated at flow level, but it stays as "Update" in Tasks level.